docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc

```diff
@@ -0,0 +1,383 @@
+
+mapConfig.setMapStoreConfig(mapStoreConfig);
+----
+--
+====
+
+|[[id-column]]`id-column`
+|Name of the column that contains the primary key.
+
+|id
+|
+
+[tabs] 
+==== 
+XML:: 
++ 
+--
+[source,xml]
+----
+<hazelcast>
+  <map name="default">
+    <map-store enabled="true">
+      <class-name>com.hazelcast.mapstore.GenericMapStore</class-name>
+      <properties>
+          <property name="external-data-store-ref">my-mysql-database</property>
+          <property name="id-column">id</property>
+      </properties>
+    </map-store>
+</hazelcast>
+----
+--
+YAML:: 
++ 
+--
+[source,yaml]
+----
+hazelcast:
+  map:
+    default:
+      map-store:
+        enabled: true
+        class-name: com.hazelcast.mapstore.GenericMapStore
+        properties:
+            external-data-store-ref: my-mysql-database
+            id-column: id
+----
+--
+Java:: 
++ 
+--
+[source,java]
+----
+MapConfig mapConfig = new MapConfig("default");
+
+MapStoreConfig mapStoreConfig = new MapStoreConfig();
+mapStoreConfig.setClassName(GenericMapStore.class.getName());
+mapStoreConfig.setProperty("external-data-store-ref", "my-mysql-database");
+mapStoreConfig.setProperty("id-column", "id");
+
+mapConfig.setMapStoreConfig(mapStoreConfig);
+----
+--
+====
+
+|[[columns]]`columns`
+|Names of the columns to map. This value must include a subset of columns in
+the table. Missing columns must have a default value defined.
+
+|
+|
+
+[tabs] 
+==== 
+XML:: 
++ 
+--
+[source,xml]
+----
+<hazelcast>
+  <map name="default">
+    <map-store enabled="true">
+      <class-name>com.hazelcast.mapstore.GenericMapStore</class-name>
+      <properties>
+          <property name="external-data-store-ref">my-mysql-database</property>
+          <property name="columns">name</property>
+      </properties>
+    </map-store>
+</hazelcast>
+----
+--
+YAML:: 
++ 
+--
+[source,yaml]
+----
+hazelcast:
+  map:
+    default:
+      map-store:
+        enabled: true
+        class-name: com.hazelcast.mapstore.GenericMapStore
+        properties:
+            external-data-store-ref: my-mysql-database
+            columns: name
+----
+--
+Java:: 
++ 
+--
+[source,java]
+----
+MapConfig mapConfig = new MapConfig("default");
+
+MapStoreConfig mapStoreConfig = new MapStoreConfig();
+mapStoreConfig.setClassName(GenericMapStore.class.getName());
+mapStoreConfig.setProperty("external-data-store-ref", "my-mysql-database");
+mapStoreConfig.setProperty("columns", "name");
+
+mapConfig.setMapStoreConfig(mapStoreConfig);
+----
+--
+====
+
+|===
+
+
+== Related Resources
+
+To monitor MapStores for each loaded entry, use the `EntryLoadedListener` interface. See the xref:events:object-events.adoc#listening-for-map-events[Listening for Map Events section] to learn how you can catch entry-based events.
+
+xref:mapstore-triggers.adoc[].
+
+== Next Steps
+
+xref:configuration-guide.adoc[Configure a map to use the generic MapStore]. The cluster does not load the MapStore until you do this.
```

Comment:
I find this a bit confusing, didn't we already configure a map store earlier in this section?

Reply:
Ah yes, thanks. I adjusted that sentence so that the next step is to see a general configuration guide for MapStores rather than as a must-do.